### PR TITLE
Make sure that Main picks up all change from Data Prepper 2.0 branch

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
### Description
_Describe what this change achieves._

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by alicejw-aws — APPROVED** [review]

File: _clients/data-prepper/index.md [hunk 1]
```diff
@@ -12,4 +12,76 @@ Data Prepper is a server side data collector capable of filtering, enriching, tr
 
 Data Prepper lets users build custom pipelines to improve the operational view of applications. Two common uses for Data Prepper are trace and log analytics. [Trace analytics]({{site.url}}{{site.baseurl}}/observability-plugin/trace/index/) can help you visualize the flow of events and identify performance problems, and [log analytics]({{site.url}}{{site.baseurl}}/observability-plugin/log-analytics/) can improve searching, analyzing and provide insights into your application.
 
+## Concepts
+
+Data Prepper is compromised of **Pipelines** that collect and filter data based on the components set within the pipeline. Each component is pluggable, enabling you to use your own custom implementation of each component. These components include:
```
Comment (cwillum) [inline _clients/data-prepper/index.md:17]: To be pedantic: "the whole comprises the parts and the parts compose the whole." (at least fix typo). 
suggestion: "...is composed of several **pipelines** that..."

Comment (Naarcha-AWS) [reply _clients/data-prepper/index.md:17]: It can be one or more pipelines though.

File: _clients/data-prepper/index.md [hunk 2]
```diff
@@ -12,4 +12,76 @@ Data Prepper is a server side data collector capable of filtering, enriching, tr
...
+Data Prepper is compromised of **Pipelines** that collect and filter data based on the components set within the pipeline. Each component is pluggable, enabling you to use your own custom implementation of each component. These components include: 
+
+- One [source](#source)
+- One or more[sinks](#sink)
+- (Optional) One [buffer](#buffer)
+- (Optional) One or more[processors](#processor)
+
+A single instance of Data Prepper can have one or more pipelines. 
+
+Each pipeline definition contains two required components **source** and **sink**. If buffers and processors are missing from the Data Prepper pipeline, Data Prepper uses the default buffer and a no-op processor.
```
Comment (cwillum) [inline _clients/data-prepper/index.md:26]: "...components: **source** and **sink**."

File: _clients/data-prepper/index.md [hunk 3]
```diff
@@ -12,4 +12,76 @@ Data Prepper is a server side data collector capable of filtering, enriching, tr
...
+- (Optional) One [buffer](#buffer)
+- (Optional) One or more[processors](#processor)
+
+A single instance of Data Prepper can have one or more pipelines. 
+
+Each pipeline definition contains two required components **source** and **sink**. If buffers and processors are missing from the Data Prepper pipeline, Data Prepper uses the default buffer and a no-op processor. 
+
+### Source 
+
+Source is the input component of a pipeline that defines the mechanism through which a Data Prepper pipeline will consume events. A pipeline can have only one source. The source can consume events either by receiving the events over HTTP or HTTPS or reading from external endpoints like OTeL Collector for traces and metrics and S3. Source have their own configuration options based on the format of the events (such as string, json,  cloudwatch logs, or open telemetry trace). The source component consumes events and writes them to the buffer component.
```
Comment (cwillum) [inline _clients/data-prepper/index.md:30]: CloudWatch. 
extra space before its mention.

File: _clients/data-prepper/index.md [hunk 4]
```diff
@@ -12,4 +12,76 @@ Data Prepper is a server side data collector capable of filtering, enriching, tr
...
+
+### Sink
+
+Sink is the output component of a pipeline that defines the destination(s) to which a Data Prepper pipeline publishes events. A sink destination could be services such as OpenSearch, S3, or another Data Prepper pipeline. When using another Data Prepper pipeline as the sink, you can chain multiple pipelines together based on the needs to the data. Sink contains it's own configurations options based on the destination type.
+
+### Processor
+
+Processors are units within the Data Prepper pipeline that can filter, transform, and enrich events into your desired format before publishing the record to the sink. The a processor is not defined in the pipeline configuration, the events publish in the format defined in the source component. You can have more than on processor within a pipeline. When using multiple processors, the processors are executed in the order they are defined inside the pipeline spec.
```
Comment (cwillum) [inline _clients/data-prepper/index.md:42]: The a processor is not defined in the pipeline configuration; rather, the events publish in the format defined in the source component.

**Review by cwillum — APPROVED** [review]
Good

**Pull request merged**